lib/platform.c

```diff
@@ -266,12 +266,14 @@ int STDCALL sf_setenv(const char *name, const char *value) {
 #endif
 }
 
-char *STDCALL sf_getenv(const char *name) {
+char * STDCALL sf_getenv(const char *name) {
 #ifdef _WIN32
-    return getenv(name);
-#else
-    return getenv(name);
-#endif
+    char* result = NULL;
+    return _dupenv_s(&result, NULL, name) ? NULL : result;
+#   else
+    char* result = getenv(name);
+    return result ? strdup(result) : NULL;
```

Comment:
who will free the buffer of strdup?

Reply:
we will leave getenv and strerror untouched. Need to free them at every invoking. Replace in the next release.